objects/src/accounts/component/template/storage/mod.rs

```diff
@@ -0,0 +1,585 @@
+use alloc::{boxed::Box, string::String, vec::Vec};
+
+use vm_core::{
+    utils::{ByteReader, ByteWriter, Deserializable, Serializable},
+    Word,
+};
+use vm_processor::{DeserializationError, Digest};
+
+mod word;
+pub use word::*;
+
+use super::AccountComponentTemplateError;
+use crate::accounts::{StorageMap, StorageSlot};
+
+mod placeholder;
+pub use placeholder::{StoragePlaceholder, StorageValue};
+
+mod init_storage_data;
+pub use init_storage_data::InitStorageData;
+
+#[cfg(feature = "std")]
+pub mod toml;
+
+// STORAGE ENTRY
+// ================================================================================================
+
+/// Represents a single entry in the component’s storage layout.
+///
+/// Each entry can describe:
+/// - A value slot (single word or multiple words).
+/// - A map slot (key-value map that occupies one storage slot).
+/// - A multi-slot entry (spanning multiple contiguous slots, with multiple values).
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StorageEntry {
+    /// A value slot, which can contain one or more words. Each word is a hex-encoded string.
+    Value {
+        /// The human-readable name of the slot.
+        name: String,
+        /// An optional description for the slot, explaining its purpose.
+        description: Option<String>,
+        /// The numeric index of this slot in the component's storage layout.
+        slot: u8,
+        /// The initial value for this slot.
+        value: WordRepresentation,
+    },
+
+    /// A map slot, containing multiple key-value pairs. Keys and values are hex-encoded strings.
+    Map {
+        /// The human-readable name of the map slot.
+        name: String,
+        /// An optional description for the slot, explaining its purpose.
+        description: Option<String>,
+        /// The numeric index of this map slot in the component's storage.
+        slot: u8,
+        /// A list of key-value pairs to initialize in this map slot.
+        map_entries: Vec<MapEntry>,
+    },
+
+    /// A multi-slot entry, representing a single logical value across multiple slots.
+    MultiSlot {
+        /// The human-readable name of this multi-slot entry.
+        name: String,
+        /// An optional description for the slot, explaining its purpose.
+        description: Option<String>,
+        /// The indices of the slots that form this multi-slot entry.
+        slots: Vec<u8>,
```

Comment:
One thing I realized is that we should be able to represent this with a `Range`. Could be done in a follow-up PR thought.